worker/uploader.go

```diff
@@ -120,8 +127,10 @@ outer:
 
 			// the uploader's contract got renewed, requeue the request
 			if errors.Is(err, errMaxRevisionReached) {
-				u.enqueue(req)
```

Comment:
a potential hotfix could be to not handle this error at all, in some cases the request would be retried using a different uploader and in the worst case the request would fail, but that's better than hanging